rollout/temlateref.go

```diff
@@ -125,6 +125,12 @@ func (r *informerBasedTemplateResolver) Resolve(rollout *v1alpha1.Rollout) error
 		return nil
 	}
 
+	// When workloadRef is resolved for the first time, TemplateResolvedFromRef = false.
+	// In this case, template must not be set
+	if !rollout.Spec.TemplateResolvedFromRef && !rollout.Spec.EmptyTemplate() {
+		return fmt.Errorf("template must be empty for workload reference rollout")
+	}
+
```

Comment:
Instead of in here, can we put this in validation library so that the linter can catch the problem?

Reply:
Updated. The method is moved to validation package. However, lint still can not catch the problem as it reports the following error. 

```
Error: spec.template.spec.containers: Required value
```

Please see the test cases I added to `validation_test.go`. During reconciliation time, since the `template.spec.containers` is filled by clientset.Get(), validation method will only report this error -`template must be empty for workload reference rollout`.

Reply:
Hi, @jessesuen , the validation are placed in both Resolve and ValidateRolloutSpec method. In `ValidateRolloutSpec`, an additional condition (`!rollout.Spec.TemplateResolvedFromRef`) is added because during reconciliation template is filled prior to validation; otherwise it results in false alarm.

An e2e test is added.